File: erp-frontend/src/types/product.ts
/** 产品 */
export interface Product {
  id: number;
  productCode: string;
  productName: string;
  categoryId: number;
  productType: number;
  spec: string;
  unit: string;
  weight: number;
  standardCost: number;
  standardPrice: number;
  status: number;
  createdAt: string;
  remark: string;
}

/** 产品详情（含图片+BOM） */
export interface ProductDetail extends Product {
  categoryName: string;
  images: ProductImage[];
  bomItems: BomItem[];
}

/** 产品表单 */
export interface ProductFormData {
  id?: number;
  productCode: string;
  productName: string;
  categoryId: number;
  productType: number;
  spec: string;
  unit: string;
  weight: number;
  standardCost: number;
  standardPrice: number;
  remark: string;
}

/** 产品查询参数 */
export interface ProductQuery {
  keyword?: string;
  categoryId?: number;
  productType?: number;
  status?: number;
  pageNum?: number;
  pageSize?: number;
}

/** 产品分类 */
export interface ProductCategory {
  id: number;
  name: string;
  parentId: number;
  sortOrder: number;
  status: number;
  children?: ProductCategory[];
}

/** 分类表单 */
export interface CategoryFormData {
  id?: number;
  name: string;
  parentId: number;
  sortOrder: number;
  remark: string;
}

/** 产品图片 */
export interface ProductImage {
  id: number;
  productId: number;
  imageUrl: string;
  sortOrder: number;
  isPrimary: number;
  createdAt: string;
}

/** BOM项 */
export interface BomItem {
  id: number;
  productId: number;
  materialId: number;
  quantity: number;
  wasteRate: number;
  sortOrder: number;
  materialName: string;
  materialCode: string;
  materialUnit: string;
  materialSpec: string;
  materialType: number;
}

/** BOM项表单 */
export interface BomItemFormData {
  materialId: number;
  quantity: number;
  wasteRate: number;
  sortOrder: number;
}

File: erp-backend/erp-product/src/main/java/com/erp/product/entity/Product.java
package com.erp.product.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.erp.common.base.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.util.List;

/**
 * 产品实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("product")
public class Product extends BaseEntity {

    /**
     * 产品编码
     */
    private String productCode;

    /**
     * 产品名称
     */
    private String productName;

    /**
     * 分类ID
     */
    private Long categoryId;

    /**
     * 产品类型（1原材料 2半成品 3成品）
     */
    private Integer productType;

    /**
     * 规格型号
     */
    private String spec;

    /**
     * 单位
     */
    private String unit;

    /**
     * 重量(kg)
     */
    private BigDecimal weight;

    /**
     * 标准成本
     */
    private BigDecimal standardCost;

    /**
     * 标准售价
     */
    private BigDecimal standardPrice;

    /**
     * 状态（0停用 1正常）
     */
    private Integer status;

    /**
     * 图片列表（非数据库字段）
     */
    @TableField(exist = false)
    private List<ProductImage> images;

    /**
     * BOM子项列表（非数据库字段）
     */
    @TableField(exist = false)
    private List<BomItem> bomItems;
}
